tests/acceptance/bootstrap/NotificationContext.php

```diff
@@ -513,20 +513,40 @@ public function userShouldHaveReceivedTheFollowingEmailFromUser(string $user, st
 		$this->assertEmailContains($user, $expectedEmailBodyContent);
 	}
 
+	/**
+	 * @Then user :user should have received the following email from user :sender ignoring extra spaces and blank lines
+	 *
+	 * @param string $user
+	 * @param string $sender
+	 * @param PyStringNode $content
+	 *
+	 * @return void
+	 * @throws Exception
+	 */
+	public function userShouldHaveReceivedTheFollowingEmailFromUserIgnoringExtraSpaces(string $user, string $sender, PyStringNode $content):void {
+		$rawExpectedEmailBodyContent = \str_replace("\r\n", "\n", $content->getRaw());
+		$expectedEmailBodyContent = $this->featureContext->substituteInLineCodes(
+			$rawExpectedEmailBodyContent,
+			$sender
+		);
+		$this->assertEmailContains($user, $expectedEmailBodyContent, true);
+	}
+
 	/***
 	 * @param string $user
 	 * @param string $expectedEmailBodyContent
+	 * @param bool $ignoreExtraSpacesAndBlankLines
 	 *
 	 * @return void
 	 * @throws GuzzleException
 	 */
-	public function assertEmailContains(string $user, string $expectedEmailBodyContent):void {
+	public function assertEmailContains(string $user, string $expectedEmailBodyContent, $ignoreExtraSpacesAndBlankLines = false):void {
 		$address = $this->featureContext->getEmailAddressForUser($user);
 		$this->featureContext->pushEmailRecipientAsMailBox($address);
 		$actualEmailBodyContent = EmailHelper::getBodyOfLastEmail($address, $this->featureContext->getStepLineRef());
 		Assert::assertStringContainsString(
-			$expectedEmailBodyContent,
-			$actualEmailBodyContent,
+			$ignoreExtraSpacesAndBlankLines ? preg_replace('/\s+/', '', $expectedEmailBodyContent) : $expectedEmailBodyContent,
```

Comment:
you could use this regex. it removes the whitespaces and empty new lines but preserves the newlines containing some text
```suggestion
			$ignoreExtraSpacesAndBlankLines ? preg_replace('/(^\s+|\s+$)/m', '', $expectedEmailBodyContent) : $expectedEmailBodyContent,
```

E.g:
```
Hi,

Alice Hansen (alice@example.org) wants to start sharing collaboration resources with you.

Please visit your federation provider and use the following details:    
  Token: %fed_invitation_token%
  ProviderDomain: https://ocis-server:9200
```
will become
```
Hi,
Alice Hansen (alice@example.org) wants to start sharing collaboration resources with you.
Please visit your federation provider and use the following details:
Token: %fed_invitation_token%
ProviderDomain: https://ocis-server:9200
```

Reply:
email has HTML formatting and other invisible characters (like multiple newlines or inline spaces) that this pattern doesn't handle. so I left first `'/\s+/'` which works correct